### PR TITLE
Update skip permission change KEP to use pvc.Status

### PR DESCRIPTION
**Pull request by gnufied**
xref https://github.com/kubernetes/enhancements/issues/695

### CODE REVIEW

File: keps/sig-storage/695-skip-permission-change/README.md [hunk 1]
```diff
@@ -70,56 +72,111 @@ But this presents following problems:
...
+     // Phase represents the current phase of PersistentVolumeClaim
+     // +optional
+     Phase PersistentVolumeClaimPhase
+     // AccessModes contains all ways the volume backing the PVC can be mounted
+     // +optional
+     AccessModes []PersistentVolumeAccessMode
+     // Represents the actual resources of the underlying volume
+     // +optional
+     Capacity ResourceList
+     // FSGroup of PVC
```
Comment (jsafrane) [inline keps/sig-storage/695-skip-permission-change/README.md:98]: The last FSGroup applied to the content of the volume.

File: keps/sig-storage/695-skip-permission-change/README.md [hunk 2]
```diff
@@ -70,56 +72,111 @@ But this presents following problems:
...
+     // + optional
+     FSGroup *int64     // <------ NEW ------
```
Comment (jsafrane) [inline keps/sig-storage/695-skip-permission-change/README.md:100]: Suggestions:
* LastFSGroup
* LastAppliedFSGroup?

File: keps/sig-storage/695-skip-permission-change/README.md [hunk 3]
```diff
@@ -70,56 +72,111 @@ But this presents following problems:
...
+  on volume will be performed by kubelet.
+
+After ownership and permissions are recursively changed, pvc.Status.FSGroup will be updated to reflect
+latest value.
+
 ### Risks and Mitigations
 
-- One of the risks is if user volume's permission was previously changed using old algorithm(which changes permission of top level directory first) and user opts in for `OnRootMismatch` `FSGroupChangePolicy` then we can't distinguish if the volume was previously only partially recursively chown'd.
+- If somehow users have volumes that have right `fsGroup` recorded in PVC's status but actual permissions on disk is different, the volume
+  could become unreadable/unwritable to the pod. This could happen if files inside volume dwerewas changed outside Kubernetes by a different process.
```
Comment (jsafrane) [inline keps/sig-storage/695-skip-permission-change/README.md:164]: typo: dwerewas

File: keps/sig-storage/695-skip-permission-change/README.md [hunk 4]
```diff
@@ -193,7 +250,7 @@ type PodSecurityContext struct {
 ### Scalability
 
 * **Will enabling / using this feature result in any new API calls?**
-  This feature should cause no new API calls.
+  This feature could trigger additional api call from kubelet to update PVC status after permission/ownership change is complete.
```
Comment (jsafrane) [inline keps/sig-storage/695-skip-permission-change/README.md:253]: It *will* trigger a single PVC.Status update call for every volume that supports fsGroup per pod with fsGroup set.

**Comment by gnufied** [comment]
closing this in favour of https://github.com/kubernetes/enhancements/pull/2024

**Pull request closed without merging**